src/Molecule.cpp

```diff
@@ -172,6 +172,104 @@ Molecule::Molecule(const std::string& xyz_filename) :
 
 
 
+/*
+ *  NAMED CONSTRUCTORS
+ */
+
+/**
+ *  @param n            the number of H atoms
+ *  @param spacing      the internuclear spacing in bohr
+ *  @param charge       the total charge
+ *
+ *  @return a charged H-chain with equal internuclear spacing
+ */
+Molecule Molecule::HChain(size_t n, double spacing, int charge) {
+
+    if (n == 0) {
+        throw std::invalid_argument("Can not create a H-chain consisting of zero atoms.");
+    }
+
+    if (spacing < 0.0) {
+        throw std::invalid_argument("Can't have a negative spacing.");
+    }
+
+
+    std::vector<Atom> h_chain;
+
+    // Put all H-atoms on a line on the x-axis: the first H is on the origin
+    double x = 0.0;  // the current x-coordinate
+    for (size_t i = 0; i < n; i++) {
+        h_chain.emplace_back(1, x, 0.0, 0.0);
+
+        x += spacing;  // proceed to the next H-atom
+    }
+
+    return Molecule(h_chain, charge);
+}
+
+
+/**
+ *  @param n            the number of H atoms
+ *  @param spacing      the internuclear spacing in bohr
+ *
+ *  @return a neutral H-chain with equal internuclear spacing
+ */
+Molecule Molecule::HChain(size_t n, double spacing) {
```

Comment:
Why did you use new function instead of charge = 0 default parameter?

Reply:
Yes, that was pretty stupid of me. I've updated this in the next commit.